### DISCUSSION TITLE
Advanced/Debug

### DISCUSSION THREAD

**Discussion by giscus** [Air-duino/document#20] [comment]
# Advanced/Debug

使用 VSCode 的Crotex-Debug插件 先决条件 安装 VSCode; 有一个支持的调试器，可以是 JLink，ST-Link，CMSIS-DAP 等，这里我们使用 DAPLink 作为例子; 安装 AirMCU 的 SDK; 安装PyOCD; VSCode 的配置 安装插件 在 VSCode 的插件市场中搜索Cortex-Debug，或者...

https://arduino.luatos.com/Advanced/Debug.html

**Comment by createskyblue** [comment]
分享一个我的调试配置
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Cortex Debug",
            "cwd": "${workspaceFolder}",
            "executable": "./build/Two_Programmable_PWM_knobs.ino.elf",
            "request": "launch",
            "type": "cortex-debug",
            "runToEntryPoint": "setup",
            "targetId": "Air001",
            "servertype": "pyocd",
            "serverpath": "C:/Users/xxx/AppData/Local/Programs/Python/Python311/Scripts/pyocd.exe",
            "armToolchainPath": "C:/Users/xxx/AppData/Local/Arduino15/packages/AirM2M/tools/xpack-arm-none-eabi-gcc/13.2.1-1.1/bin",
            "gdbPath": "C:/Users/xxx/AppData/Local/Arduino15/packages/AirM2M/tools/xpack-arm-none-eabi-gcc/13.2.1-1.1/bin/arm-none-eabi-gdb.exe",
        }
    ]
}
 
```

**Reply by YangX1hao** [reply]
牛哇大佬  按照大佬的配置成功调试

**Comment by MirrorCY** [comment]
用 stlink-v2 成功调试，使用了 @createskyblue 的配置模板，额外进行了 libusb 驱动的替换
- 下载 [Zadig](https://zadig.akeo.ie/)
- 显示所有设备 ![image](https://github.com/Air-duino/document/assets/37006258/f9cc704d-2cbd-4dae-8741-45bd2bedb575)
- 替换成 libusb-win32 驱动 ![image](https://github.com/Air-duino/document/assets/37006258/e7e4159b-2a1e-417f-9487-25d65f8c29ce)
